src/routes/firmware.ts

```diff
@@ -0,0 +1,20 @@
+import { Router } from 'express';
+import {
+  errorSeen,
+  installFirmware,
+  message,
+  SUCCESS_MESSAGE,
+} from 'util/firmware';
+const router = Router();
+
+router.get('/install', async (req, res) => {
+  const firmwareFile = req?.body?.fileName || '';
+  const resp = installFirmware(firmwareFile);
+  res.json(resp);
+});
+
+router.get('/progress', async (req, res) => {
+  res.json({ isRunning: message !== SUCCESS_MESSAGE, errorSeen });
```

Comment:
let's this endpoint the same way: the logic doesnt live here 🙏 , something like: 
`res.json(getInstallationProgress())`

and created the method in the firmware util